### ISSUE TITLE
Opentsdb Security

### ISSUE THREAD

**Issue by MohamedHz** [OpenTSDB/opentsdb#667]
I am successfully running the opentsdb with port 4242 and gathering the metrics from system. But  the issue we are facing practically was and who-ever can gathered the metrics from TSDB server by using port 4242 and draw the graphs.

Please share me the good security tips for securing the port 4242 and metrics should be gathered if only by authenticating the port by any system like ( For eg : Graphana ) not by simply typing the url : 192.168.1.1:4242

Awaiting for your reply!

**Comment by trixpan** [comment]
@MohamedHz 

OpenTSDB has no security built in. Whoever can read it will also be able to write to it.

Cheers

**Comment by mgoralczyk-viasat** [comment]
If you have 2.1+ you can set the tsd.mode setting to "ro" to make a TSD read-only, then with firewall settings make only the read-only TSDs accessible to the right people (or servers) for reading.

**Comment by pau** [comment]
We delegate security to Nginx and use Basic HTTP Authentication.

```
location /tsdb/ {
        auth_basic "Restricted";
        auth_basic_user_file /etc/nginx/tsdb.htpasswd;
        proxy_pass http://localhost:4242/;
    }
```

Note that you should set `tsd.network.bind` to `127.0.0.1` so external requests to port 4242 can't get thru.

**Comment by IDerr** [comment]
This one is resolved @manolama #899
You can close this :D
